services/chronikService.ts

```diff
@@ -387,22 +387,58 @@ export class ChronikBlockchainClient {
     }
   }
 
-  private getSortedInputAddresses (transaction: Tx): string[] {
+  private getSortedInputAddresses (transaction: Tx): Array<{address: string, amount: Prisma.Decimal}> {
     const addressSatsMap = new Map<string, bigint>()
-
     transaction.inputs.forEach((inp) => {
       const address = outputScriptToAddress(this.networkSlug, inp.outputScript)
       if (address !== undefined && address !== '') {
         const currentValue = addressSatsMap.get(address) ?? 0n
         addressSatsMap.set(address, currentValue + inp.sats)
       }
     })
-
     const sortedInputAddresses = Array.from(addressSatsMap.entries())
       .sort(([, valueA], [, valueB]) => Number(valueB - valueA))
-      .map(([address]) => address)
+    return sortedInputAddresses.map(([address, sats]) => {
+      const format = xecaddr.detectAddressFormat(address)
+      // satoshisToUnit returns Promise; but we can't make this function async easily without refactor, so keep raw satoshis division inline
+      const decimal = new Prisma.Decimal(sats.toString())
+      let amount: Prisma.Decimal
+      if (format === xecaddr.Format.Xecaddr) {
+        amount = decimal.dividedBy(1e2)
+      } else if (format === xecaddr.Format.Cashaddr) {
+        amount = decimal.dividedBy(1e8)
+      } else {
+        amount = decimal
+      }
+      return { address, amount }
+    })
+  }
 
-    return sortedInputAddresses
+  private getSortedOutputAddresses (transaction: Tx): Array<{address: string, amount: Prisma.Decimal}> {
+    const addressSatsMap = new Map<string, bigint>()
+    transaction.outputs.forEach((out) => {
+      const address = outputScriptToAddress(this.networkSlug, out.outputScript)
+      if (address !== undefined && address !== '') {
+        const currentValue = addressSatsMap.get(address) ?? 0n
+        addressSatsMap.set(address, currentValue + out.sats)
+      }
+    })
+    const sortedOutputAddresses = Array.from(addressSatsMap.entries())
+      .sort(([, valueA], [, valueB]) => Number(valueB - valueA))
+      .map(([address, sats]) => {
+        const format = xecaddr.detectAddressFormat(address)
```

Comment:
Same thing here. No need to detect the network per tx address, it will always be the same one (`this.networkSlug`)